agagd/agagd_core/views/players_profile.py

```diff
@@ -45,56 +45,68 @@ def get(self, request, *args, **kwargs):
             Q(pin_player__exact=player_id)
         ).values("pin_player", "rating", "sigma")
 
-        # compute additional tables for opponents & tournament info. here
-        # TODO: refactor this into something nicer.
-        opponent_data = {}
-        tourney_data = {}
-        for game in player_games:
-            try:
-                t_dat = tourney_data.get(game.tournament_code.pk, {})
-                t_dat["tournament"] = game.tournament_code
-                t_dat["won"] = t_dat.get("won", 0)
-                t_dat["lost"] = t_dat.get("lost", 0)
-
-                # Set default game_date to None
-                game_date = None
-
-                # Check for 0000-00-00 dates
```

Comment:
The purpose of this check is not entirely clear;  supposedly all games should have a valid date?  This piece of logic, which leaves the game_date as `None` in selected cases, is not reflected in the new implementation.

Reply:
the records in the games database are not clean or unified.  generally a good idea to assume that the checks are because there's bad data in prod that are >20 years old :)  I'll see if i can run some queries.

Reply:
very fair, but I'm still a bit confused as to how what exactly this check is doing;

the Game table uses DateField to represent dates, which returns `datetime.date` objects.  So this comparison against a string would always return True, meaning that the field is always used anyway.. Further, you cannot create a `datetime.date(0, 0, 0)' object to begin with, as the year/month/day combination is out of range, so it's not like there's any chance that any of the dates actually match '0000-00-00'.  I'm a bit skeptical this check is doing anything at all;  if there's a way for you to access the database and find some examples of problematic game dates, that'll help clear a light into what's going on here, and how I can integrate the special cases in the current query.